source/common/network/socket_option_impl.h

```diff
@@ -123,17 +126,20 @@ class SocketOptionImpl : public Socket::Option, Logger::Loggable<Logger::Id::con
    * @param socket the socket on which to apply the option.
    * @param optname the option name.
    * @param value the option value.
+   * @param size the option value size.
    * @return a Api::SysCallIntResult with rc_ = 0 for success and rc = -1 for failure. If the call
    * is successful, errno_ shouldn't be used.
    */
   static Api::SysCallIntResult setSocketOption(Socket& socket,
                                                const Network::SocketOptionName& optname,
-                                               absl::string_view value);
+                                               const void* value, size_t size);
```

Comment:
Why not `const std::vector<uint8_t>& value)`?

Reply:
to match what the syscall is taking. a const ref of vector must be a vector instance but here it want a memory address and a size.

Reply:
It wasn't doing that before, so this adds an unnecessary change to a fix, IMHO.

It's a nit, anyway, I was going back-and-forth myself between both variants in #7969, so I wanted to see how did you justify your choice.

Reply:
Yeah it is not necessary though the underlying type is no longer std::string thus I didn't feel string_view is appropriate. So just made this same as the syscall.

Reply:
Erm, I was suggesting `const std::vector<uint8_t>& value`, not `absl::string_view value`.

Reply:
because then it must be a vector instance, but not a (address, size) pair, which is not a big deal really though